[*] Start mutation process:
   - targets: factorial.py
   - tests: ts-3-5/test_3-5_0-8_254.py
[*] 3 tests passed:
   - test_3-5_0-8_254 [0.12508 s]
[*] Start mutants generation and execution:
   - [#   1] AOR factorial: 
--------------------------------------------------------------------------------
  14:     
  15:     if n <= 0:
  16:         return 0
  17:     
- 18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
+ 18:     return reduce((lambda x, y: (x / y)), range(1, n + 1))
--------------------------------------------------------------------------------
[0.14910 s] killed by ts-3-5/test_3-5_0-8_254.py::test_factorial_positive_numbers

def test_factorial_positive_numbers():
        assert factorial(1) == 1
>       assert factorial(5) == 120
E       AssertionError

ts-3-5/test_3-5_0-8_254.py:17: AssertionError
   - [#   2] AOR factorial: 
--------------------------------------------------------------------------------
  14:     
  15:     if n <= 0:
  16:         return 0
  17:     
- 18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
+ 18:     return reduce((lambda x, y: (x // y)), range(1, n + 1))
--------------------------------------------------------------------------------
[0.15265 s] killed by ts-3-5/test_3-5_0-8_254.py::test_factorial_positive_numbers

def test_factorial_positive_numbers():
        assert factorial(1) == 1
>       assert factorial(5) == 120
E       AssertionError

ts-3-5/test_3-5_0-8_254.py:17: AssertionError
   - [#   3] AOR factorial: 
--------------------------------------------------------------------------------
  14:     
  15:     if n <= 0:
  16:         return 0
  17:     
- 18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
+ 18:     return reduce((lambda x, y: (x ** y)), range(1, n + 1))
--------------------------------------------------------------------------------
[0.15291 s] killed by ts-3-5/test_3-5_0-8_254.py::test_factorial_positive_numbers

def test_factorial_positive_numbers():
        assert factorial(1) == 1
>       assert factorial(5) == 120
E       AssertionError

ts-3-5/test_3-5_0-8_254.py:17: AssertionError
   - [#   4] AOR factorial: 
--------------------------------------------------------------------------------
  14:     
  15:     if n <= 0:
  16:         return 0
  17:     
- 18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
+ 18:     return reduce((lambda x, y: (x * y)), range(1, n - 1))
--------------------------------------------------------------------------------
[0.14166 s] incompetent

TypeError: factorial:18: TypeError: reduce() of empty iterable with no initial value

   - [#   5] COI factorial: 
--------------------------------------------------------------------------------
  11:         The factorial number for n (n!).  Input
  12:         of 0 or less will output 0.
  13:     '''
  14:     
- 15:     if n <= 0:
+ 15:     if not (n <= 0):
  16:         return 0
  17:     
  18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
--------------------------------------------------------------------------------
[0.14395 s] incompetent

TypeError: factorial:18: TypeError: reduce() of empty iterable with no initial value

   - [#   6] ROR factorial: 
--------------------------------------------------------------------------------
  11:         The factorial number for n (n!).  Input
  12:         of 0 or less will output 0.
  13:     '''
  14:     
- 15:     if n <= 0:
+ 15:     if n >= 0:
  16:         return 0
  17:     
  18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
--------------------------------------------------------------------------------
[0.14818 s] incompetent

TypeError: factorial:18: TypeError: reduce() of empty iterable with no initial value

   - [#   7] ROR factorial: 
--------------------------------------------------------------------------------
  11:         The factorial number for n (n!).  Input
  12:         of 0 or less will output 0.
  13:     '''
  14:     
- 15:     if n <= 0:
+ 15:     if n < 0:
  16:         return 0
  17:     
  18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
--------------------------------------------------------------------------------
[0.14654 s] incompetent

TypeError: factorial:18: TypeError: reduce() of empty iterable with no initial value

   - [#   8] SDL factorial: 
--------------------------------------------------------------------------------
  12:         of 0 or less will output 0.
  13:     '''
  14:     
  15:     if n <= 0:
- 16:         return 0
+ 16:         pass
  17:     
  18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
--------------------------------------------------------------------------------
[0.14276 s] incompetent

TypeError: factorial:18: TypeError: reduce() of empty iterable with no initial value

   - [#   9] SDL factorial: 
--------------------------------------------------------------------------------
  14:     
  15:     if n <= 0:
  16:         return 0
  17:     
- 18:     return reduce((lambda x, y: (x * y)), range(1, n + 1))
+ 18:     pass
--------------------------------------------------------------------------------
[0.14395 s] killed by ts-3-5/test_3-5_0-8_254.py::test_factorial_positive_numbers

def test_factorial_positive_numbers():
>       assert factorial(1) == 1
E       AssertionError

ts-3-5/test_3-5_0-8_254.py:16: AssertionError
[*] Mutation score [1.55017 s]: 100.0%
   - all: 9
   - killed: 4 (44.4%)
   - survived: 0 (0.0%)
   - incompetent: 5 (55.6%)
   - timeout: 0 (0.0%)
